source/synthDriverHandler.py

```diff
@@ -448,29 +459,42 @@ def setSynth(name, isFallback=False):
 	else:
 		prevSynthName = None
 	try:
-		_curSynth = getSynthInstance(name)
+		_curSynth = getSynthInstance(name, asDefault)
+	except:  # noqa: E722 # Legacy bare except
+		log.error(f"setSynth failed for {name}", exc_info=True)
+	
+	if _curSynth is not None:
 		_audioOutputDevice = config.conf["speech"]["outputDevice"]
 		if not isFallback:
 			config.conf["speech"]["synth"] = name
-		log.info("Loaded synthDriver %s" % name)
+		log.info(f"Loaded synthDriver {_curSynth.name}")
 		return True
-	except:  # noqa: E722 # Legacy bare except
-		log.error("setSynth", exc_info=True)
-		# As there was an error loading this synth:
-		if prevSynthName:
-			# There was a previous synthesizer, so switch back to that one.
-			setSynth(prevSynthName, isFallback=True)
-		else:
-			# There was no previous synth, so fallback to the next available default synthesizer
-			# that has not been tried yet.
-			try:
-				nextIndex = defaultSynthPriorityList.index(name) + 1
-			except ValueError:
-				nextIndex = 0
-			if nextIndex < len(defaultSynthPriorityList):
-				newName = defaultSynthPriorityList[nextIndex]
-				setSynth(newName, isFallback=True)
-		return False
+	# As there was an error loading this synth:
+	elif prevSynthName:
+		log.info(f"Falling back to previous synthDriver {prevSynthName}")
+		# There was a previous synthesizer, so switch back to that one.
+		setSynth(prevSynthName, isFallback=True)
+	else:
+		# There was no previous synth, so fallback to the next available default synthesizer
+		# that has not been tried yet.
+		log.info(f"Searching for next synthDriver")
+		findAndSetNextSynth(name)
+	return False
+
+
+def findAndSetNextSynth(currentSynthName: str) -> bool:

```

Comment:
Can you take a look to see if it is possible to add tests for the synth driver handler?

Reply:
The way `speechManagerTestHarness` works make this hard to do in a meaningful way. I started writing a testing outline but it would require a rework (with unknown scope) to avoid `speechManagerTestHarness` breaking `synthdriverHandler`. I think it might be worth tracking this as something to do in the future.